src/tree/scaling/ScaleItem.ts

```diff
@@ -43,22 +47,48 @@ export class ScaleItem implements ContainerAppsItem {
     getTreeItem(): TreeItem {
         return {
             id: this.id,
-            label: localize('scaling', 'Scaling'),
+            label: this.hasUnsavedChanges() ? `${scaling}*` : scaling,
             contextValue: ScaleItem.contextValue,
             iconPath: treeUtils.getIconPath('scaling'),
             collapsibleState: TreeItemCollapsibleState.Collapsed,
         }
     }
 
-    async getChildren?(): Promise<TreeElementBase[]> {
+    getChildren(): TreeElementBase[] {
+        let scale: Scale | undefined;
+
+        if (this.hasUnsavedChanges()) {
```

Comment:
Don't need it now, but maybe just a side-note. Anytime we have an item that is part of the "batch changes", we could have a base model for it that has the same implementation for how to get the value/children and check for unsavedChanges, etc.

Reply:
Yeah, I totally get what you mean, I was thinking about doing something like this eventually as well.  I'm not sure if I prefer extending from a base here or just implementing an interface (leaning towards the latter).  If we ever decide we want to implement `discard changes` on the item level, that would also be a good candidate to include as well.